bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -136,6 +138,12 @@ bool PreProcessor::checkClientMsgCorrectness(const ClientPreProcessReqMsgUniqueP
 template <>
 void PreProcessor::onMessage<ClientPreProcessRequestMsg>(ClientPreProcessRequestMsg *msg) {
   if (preProcessorMetrics_.requestReceived.Get().Get() % 10 == 0) metricsComponent_.UpdateAggregator();
+  auto currTime = std::chrono::duration_cast<std::chrono::seconds>(std::chrono::steady_clock::now().time_since_epoch());
```

Comment:
All the piece of code you added should be moved under if (preProcessorMetrics_.requestReceived.Get().Get() % 10 == 0) condition to avoid an overhead of performing this check 1000 per sec.

Reply:
Done